AEPCore/Tests/EventHubTests/SDKInstanceIdentifierTests.swift

```diff
@@ -0,0 +1,194 @@
+        "\u{0085}", // newline character
+        "\u{2028}", // newline character
+        "\u{2029}", // newline character
+        "\u{0009}", // whitespace character
+        ]
+    
+    override func setUp() {
+    }
+    
+    func testDefaultInitializedFromEnum() {
+        let instance: SDKInstanceIdentifier = .default
+        
+        XCTAssertEqual(nil, instance.id)
+        XCTAssertEqual("default-instance", instance.description)
+    }
+    
+    func testDefaultInitializedFromInit() {
+        guard let instance = SDKInstanceIdentifier(id: "default-instance") else {
+            XCTFail("SDKInstanceIdentifier init returned nil.")
+            return
+        }
+        
+        XCTAssertEqual(nil, instance.id)
+        XCTAssertEqual("default-instance", instance.description)
+    }
+    
+    func testIDInitializedFromEnum() {
+        let instance: SDKInstanceIdentifier = .id("happy")
+        
+        XCTAssertEqual("happy", instance.id)
+        XCTAssertEqual("happy", instance.description)
+    }
+    
+    func testIDInitializedFromInit() {
+        guard let instance = SDKInstanceIdentifier(id: "happy") else {
+            XCTFail("SDKInstanceIdentifier init returned nil.")
+            return
+        }
+        
+        XCTAssertEqual("happy", instance.id)
+        XCTAssertEqual("happy", instance.description)
+    }
+    
+    func testEmptyStringFailsInitialization() {
+        XCTAssertNil(SDKInstanceIdentifier(id: ""))
+    }
+    
+    func testEmptyStringInitalizedFromEnum() {
+        let instance: SDKInstanceIdentifier = .id("")
+        
+        XCTAssertEqual("", instance.id)
+        XCTAssertEqual("", instance.description)
+    }
+    
+    func testStringExtensionInstanceAwareName() {
+        let instance: SDKInstanceIdentifier = .id("tenant")
+        let string = "message"
+        XCTAssertEqual("message-tenant", string.instanceAwareName(for: instance))
+    }
+    
+    func testStringExtensionInstanceAwareNameDefault() {
+        let instance: SDKInstanceIdentifier = .default
+        let string = "message"
+        XCTAssertEqual("message", string.instanceAwareName(for: instance))
+    }
+    
+    func testStringExtensionInstanceAwareFilename() {
+        let instance: SDKInstanceIdentifier = .id("tenant")
+        let string = "filename"
+        XCTAssertEqual("aep.tenant.filename", string.instanceAwareFilename(for: instance))
+    }
+    
+    func testStringExtensionInstanceAwareFilenameDefault() {
+        let instance: SDKInstanceIdentifier = .default
+        let string = "filename"
+        XCTAssertEqual("filename", string.instanceAwareFilename(for: instance))
+    }
+    
+    func testListOfValidNamesPassInitialization() {
+        for i in validIdentifiers {
+            XCTAssertNotNil(SDKInstanceIdentifier(id: i), "SDKInstanceIdentifier init returned nil for id '\(i)' but expected to pass.")
+        }
+    }
+    
+    func testListOfValidNamesPassFileCreation() {
```

Comment:
Can you remove this test which creates files? We can add something like this as part of the new ServiceProvider implementation.